internal/backends/java/java.go

```diff
@@ -0,0 +1,201 @@
+// Package java provides a backend for Java using maven.
+package java
+
+import (
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/replit/upm/internal/api"
+	"github.com/replit/upm/internal/util"
+)
+
+type Dependency struct {
+	XMLName    xml.Name `xml:"dependency"`
+	GroupId    string   `xml:"groupId"`
+	ArtifactId string   `xml:"artifactId"`
+	Version    string   `xml:"version"`
+}
+
+type Project struct {
+	XMLName      xml.Name     `xml:"project"`
+	ModelVersion string       `xml:"modelVersion"`
+	GroupId      string       `xml:"groupId"`
+	ArtifactId   string       `xml:"artifactId"`
+	Version      string       `xml:"version"`
+	Dependencies []Dependency `xml:"dependencies>dependency"`
+}
+
+var emptyProject = Project{
+	ModelVersion: "4.0.0",
+	GroupId:      "co.repl",
```

Comment:
hmm, unsure what the default value should be but people using upm outside of replit might not want co.repl

Reply:
going with `mygroupid` and `myartifactid`